src/engine/ExportQueryExecutionTrees.cpp

```diff
@@ -400,12 +400,15 @@ ExportQueryExecutionTrees::handleIriOrLiteral(
   AD_CORRECTNESS_CHECK(word.isLiteral());
   if (onlyReturnLiteralsWithXsdString) {
     if (isPlainLiteralOrLiteralWithXsdString(word)) {
+      if (word.hasDatatype()) {
+        word.getLiteral().removeDatatypeOrLanguageTag();
+      }
       return std::move(word.getLiteral());
     }
     return std::nullopt;
   }
 
-  if (word.hasDatatype() && !isPlainLiteralOrLiteralWithXsdString(word)) {
+  if (word.hasDatatype() || word.hasLanguageTag()) {
```

Comment:
if the `removeDatatypeOrLanguageTag` function also does the right thing if there is no datatype or language tag (verify this by a unit test if not already the case), 
then you can get rid of the `if` and just comment that this always works.